langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -579,6 +579,20 @@ function testReplace() {
     var regExpr5 = re `0+`;
     string result6 = regExpr5.replace(str6, replacementFunctionForReplace, 4);
     assertEquality("1001711", result6);
+
+    string str7 = "🔜 #RayoBarça 🔵🔴 https://t.co/iHDUx7EmFJ";
+    var regExpr6 = re `\p{S}`;
+    string result7 = regExpr6.replace(str7, replacementFunctionForReplace);
+    assertEquality("2 #RayoBarça 🔵🔴 https://t.co/iHDUx7EmFJ", result7);
```

Comment:
The symbol should be replace with value 1.
In the user point of view symbol is a one character. So the end index should be one in the replacement function.

In here, I think this gives 2 because symbol is represent by two code points, but Imo, when we give the results to users, we should consider it as one